docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,698 @@
+
+![Before using the non-interactive defaults](./assets/before.png "Before requiring that all commits consist of subtree roots")
+![after](./assets/after.png "After requiring that all commits consist of subtree roots")
+
+Below is an example block that has been filled using the non-interactive default rules to ensure that all commitments consist only of subtree roots.
+
+![example](./assets/example-full-block.png "example")
+
+- `ns = 1`: The first namespace is put in the beginning of the messages space.
+- `ns = 2`: Starting at the largest power of 2 that is not larger than the message length or `k`.
+- `ns = 3`: Message spanning multiple rows. So, it will be at the beginning of a new row.
+- `ns  = 4`: Starting at the largest power of 2 that is not larger than the message length or `k`.
+
+Not only does doing this allow for easy trust minimized message inclusion checks for specific messages by light clients, but also allows for the creation of message inclusion fraud proofs for all messages in the block. This is important to reduce the trust assumptions made by light clients.
+
+## Decision
+
+TBD
+
+While there certainly can be some decisions here, whether or not we begin following the non-interactive defaults isn't really the goal of this ADR. Please see the note at the top of this document.
+
+## Alternative Designs
+
+While all commitments signed over must only consist of subtree roots, its worth noting that non-interactive defaults are just that, defaults! It's entirely possible that block producers use some mechanism to notify the signer of the commitments that they must sign over, or even that the block producers are signing the transactions paying for the inclusion of the message on behalf of the users. This would render the non-interactive defaults, and the padding accompanied by them, to not be necessary. Other solutions are not mutually exclusive to non-interactive defaults, and do not even have to be built by the core team, so covering those solutions in a more in depth way is outside the scope of this ADR.
+
+However, the default implementation of non-interative defaults is within the scope of this ADR. Whatever design we use ultimately needs to support not using the non-interactive defaults. Meaning we should be able to encode block data into a square even if the messages are not arranged according to the non-interactive defaults. Again, this does not change the requirement that all share commitments signed over in PFDs consist only of subtree roots.
+
+## Detailed Design
+
+To recap the default constraints of arranging a square:
+
+- All messages must be ordered lexicographically by namespace.
+- The commitments signed over in each `MsgPayForData` must consist only of subtree roots of the data square.
+- If a `MsgPayForData` is added to the square, then its corresponding message must also be included.
+- There must not be a message without a `MsgPayForData` (does this need to be a rule? cc @adlerjohn).
```

Comment:
I think having this rule at the very least makes things simpler, even if not strictly required (have to think of if it's required). So, yes.

Ditto for the repeat of this line further down.

Reply:
@evan-forbes can you remove uses of `@adlerjohn` in this page?

Reply:
yeah sure thing https://github.com/celestiaorg/celestia-app/pull/673/commits/eb745c89db1feca139b77ba017303b34587b4ca2
good catch, forgot about this one